include/op.h

```diff
@@ -105,11 +101,12 @@ typedef struct header_s {
 typedef struct champion
 {
     header_t *champ_header;       // header
-    int id;                       // id of the champ
-    int num_instuctions;          // number of instructions
-    op_t *instructions;           // instruction array
-    int registers[16];            // address of registers
-    int pc;                       // program counter
+    int id;                       // id of champ
+    int address;                  // address of champ
+    int num_inst;                 // number of instructions
+    unsigned long long *inst;     // instruction array
```

Comment:
The instruction array may not need to be unsigned long long. I'm experimenting with this data structure.